backend/scripts/test-integration.sh

```diff
@@ -1,52 +1,132 @@
 #!/bin/bash
-# Run integration tests with PostgreSQL
+# Run integration tests with PostgreSQL in Docker
+# This script creates a temporary PostgreSQL instance for testing
 
 set -e
 
-echo "Starting PostgreSQL container..."
+# Colors for output
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+echo -e "${BLUE}PostgreSQL Integration Test Runner${NC}"
+echo "=================================="
+
+# Configuration
+CONTAINER_NAME="mnemosyne-test-db-$$"  # Use PID to avoid conflicts
+TEST_PORT=15432  # Use non-standard port to avoid conflicts
+DB_USER="postgres"
+DB_PASSWORD="postgres"
+DB_NAME="mnemosyne_test"
+
+# Cleanup function
+cleanup() {
+    echo -e "\n${YELLOW}Cleaning up...${NC}"
+    docker stop $CONTAINER_NAME >/dev/null 2>&1 || true
+    docker rm $CONTAINER_NAME >/dev/null 2>&1 || true
+}
+
+# Set trap to cleanup on exit
+trap cleanup EXIT
 
 # Check if docker is installed
 if ! command -v docker &> /dev/null; then
-    echo "Docker is required but not installed. Please install Docker."
+    echo -e "${RED}Docker is required but not installed. Please install Docker.${NC}"
     exit 1
 fi
 
+# Check if container already exists and remove it
+if docker ps -a | grep -q $CONTAINER_NAME; then
+    echo "Removing existing test container..."
+    cleanup
+fi
+
 # Start PostgreSQL container
+echo -e "${BLUE}Starting PostgreSQL container on port $TEST_PORT...${NC}"
 docker run -d \
-  --name mnemosyne-test-db \
-  -e POSTGRES_USER=mnemosyne_test \
-  -e POSTGRES_PASSWORD=test_password \
-  -e POSTGRES_DB=mnemosyne_test \
-  -p 5432:5432 \
-  postgres:15
+  --name $CONTAINER_NAME \
+  -e POSTGRES_USER=$DB_USER \
+  -e POSTGRES_PASSWORD=$DB_PASSWORD \
```

Comment:
Good timeout implementation! The 30-second timeout prevents CI/CD builds from hanging. The progressive dot output gives visual feedback that the script is still running.

Consider adding an exponential backoff or checking Docker logs if the database fails to start - this could help diagnose issues in CI environments.